lib/mongo/libmongocrypt.rb

```diff
@@ -0,0 +1,19 @@
+# Copyright (C) 2019 MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module Mongo
+  class Libmongocrypt
+    autoload(:Binding, 'mongo/libmongocrypt/binding.rb')
```

Comment:
This is one possible way to have `Binding` only load when it's called. That way, if there's no libmongocrypt (path specified by an env variable for now, maybe a separate gem later), the user gets a specific error message when they try to use `Mongo::Libmongocrypt::Binding` rather than an error message when loading the entire driver.

Another way we could do this is by raising an error in the class and then catching it on `require` in `mongo.rb`, for example:
```
# in mongo.rb

begin
  require 'mongo/libmongocrypt'
rescue LibmongocryptError => e
  warn('Could not load Mongo::Libmongocrypt::Binding because...`)
end
```
I've honestly never used a pattern like this before in Ruby so I'm happy to chat more about tradeoffs or better ways to do it.